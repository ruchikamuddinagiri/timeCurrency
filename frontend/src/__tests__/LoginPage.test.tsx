import * as React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";
import LoginPage from '../pages/LoginForm';


jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}));

describe('MyComponent', () => {
  const mockNavigate = jest.fn();
  

  it('renders correctly', () => {

    const { container } = render(<MemoryRouter>
      <LoginPage />
    </MemoryRouter>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
