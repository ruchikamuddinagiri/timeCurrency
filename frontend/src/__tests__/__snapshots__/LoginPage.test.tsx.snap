// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent renders correctly 1`] = `
<section
  class="bg-ct-black-600 min-h-screen grid place-items-center"
>
  <div
    class="w-full"
  >
    <h1
      class="text-4xl xl:text-6xl text-center font-[600] text-ct-black-600 mb-4"
    >
      LOGIN
    </h1>
    <form
      class="max-w-md w-full mx-auto overflow-hidden shadow-lg bg-ct-dark-200 squared-2xl p-8 space-y-5"
    >
      <div
        class="my-3"
      >
        <label
          class="block text-gray-600 mb-1 font-medium"
          for="email"
        >
          Email
        </label>
        <input
          class="block w-full py-2 px-4 rounded-md bg-gray-100 border-transparent focus:outline-none focus:ring-2 focus:ring-blue-500 focus:bg-white focus:border-transparent"
          id="email"
          name="email"
          placeholder=""
          type="email"
        />
      </div>
      <div
        class="my-3"
      >
        <label
          class="block text-gray-600 mb-1 font-medium"
          for="password"
        >
          Password
        </label>
        <input
          class="block w-full py-2 px-4 rounded-md bg-gray-100 border-transparent focus:outline-none focus:ring-2 focus:ring-blue-500 focus:bg-white focus:border-transparent"
          id="password"
          name="password"
          placeholder=""
          type="password"
        />
      </div>
      <div
        class="text-right"
      >
        <a
          class=""
          href="/forgotpassword"
        >
          Forgot Password?
        </a>
      </div>
      <button
        class="w-full py-3 font-semibold rounded-lg outline-none border-none flex justify-center ct-black-600 false"
        type="submit"
      >
        <span>
          Login
        </span>
      </button>
      <span
        class="block"
      >
        New User?
         
        <a
          class="text-ct-blue-600"
          href="/register"
        >
          Sign Up Here
        </a>
      </span>
    </form>
  </div>
</section>
`;
